src/dc_tools.rs

```diff
@@ -50,24 +46,19 @@ pub unsafe fn dc_strdup_keep_null(s: *const libc::c_char) -> *mut libc::c_char {
 }
 
 pub unsafe fn dc_atoi_null_is_0(s: *const libc::c_char) -> libc::c_int {
-    return if !s.is_null() { atoi(s) } else { 0i32 };
+    if !s.is_null() {
+        to_str(s).parse().unwrap_or_default()
+    } else {
+        0
+    }
 }
 
-pub unsafe fn dc_atof(str: *const libc::c_char) -> libc::c_double {
-    // hack around atof() that may accept only `,` as decimal point on mac
-    let test: *mut libc::c_char = dc_mprintf(b"%f\x00" as *const u8 as *const libc::c_char, 1.2f64);
-    *test.offset(2isize) = 0i32 as libc::c_char;
-    let mut str_locale: *mut libc::c_char = dc_strdup(str);
-    dc_str_replace(
-        &mut str_locale,
-        b".\x00" as *const u8 as *const libc::c_char,
-        test.offset(1isize),
-    );
-    let f: libc::c_double = atof(str_locale);
-    free(test as *mut libc::c_void);
-    free(str_locale as *mut libc::c_void);
+pub fn dc_atof(s: *const libc::c_char) -> libc::c_double {
+    if s.is_null() {
+        return 0.;
+    }
 
-    f
+    to_str(s).parse().unwrap_or_default()
```

Comment:
And I guess automatically converts to double here.

Reply:
yes